src/index.ts

```diff
@@ -72,25 +110,32 @@ function _hydrate(
     shouldExpandOnMouseOver,
     textMode = TextMode.Text,
     htmlContainerTag = "span",
+    specialCharacters = {},
   } = config;
+
   let lineText = line.text;
 
   function createLineNode(lineText: string) {
     switch (textMode) {
       case TextMode.Text:
-        // TODO: move this special character replacement functionality into a general purpose config
-        // passed in at the top-level.
-        if (lineText.includes("@Q ")) {
-          // if this is a quotation, turn it into a <blockquote> element
-          const [before, quote] = lineText.split("@Q ", 2);
-          const pre = document.createTextNode(before);
-          node.appendChild(pre);
-          const el = document.createElement("blockquote");
-          el.innerText = quote;
-          return el;
-        } else {
-          return document.createTextNode(lineText);
+        for (const [specialCharRegex, replacementFn] of Object.entries({
+          ...DefaultReplacements,
+          ...specialCharacters
+        })) {
+          const matches = lineText.match(specialCharRegex)
+          if (matches) {
+            const container = document.createElement(htmlContainerTag);
+            const [wholeMatch, innerMatchText, _] = matches
+            // compute pre and post match text
+            const [pre, post] = lineText.split(wholeMatch)
+            container.appendChild(createLineNode(pre))
```

Comment:
oh interesting are you calling recursively so it does all matches? and you're just looking at the first match here?

Reply:
yeah so you can have bold/emphasis in a blockquote for example